src/Common/TagStoreInterface.php

```diff
@@ -4,15 +4,66 @@
 
 interface TagStoreInterface
 {
-    public function tagObject($objectId, array $tags);
+    /**
+     * Associate the given tags to cache item identified by $itemId.
+     *
+     * Example: associate a HTTP cache identifier (e.g. a content-digest in the
+     *          case of the Symfony HTTP cache) with a set of tags.
+     *
+     * @param integer $itemId
+     * @param string[] $tags
+     */
+    public function tagItemId($itemId, array $tags);
 
-    public function untagObject($objectId, array $tags);
+    /**
+     * Unassociate the given tags from the cache item identified by $itemId.
+     *
+     * NOTE: This is difficult for an so-called "cache based" implementation, is it necessary?
+     *
+     * @param integer $itemId
+     * @param string[] $tags
+     */
+    public function untagItemId($itemId, array $tags);
```

Comment:
Well, you can solve this by having both one file per item and one file per tag, the item-file listing the tags of a given item, and the tag file listing the items with a given tag. It's less efficient for tagging though.

The bare minimum of methods I'd consider necessary for an HTTP cache is probably

* tagItemId
* getItemIdsWithTags
* removeTags
* removeItemIds

Let me go a bit further into `removeItemIds`. Consider the following scenario:

1. A new resource is created, http://www.example.com/foo
2. A client accesses this resource, it's cached and tagged with a, b, c.
3. A site editor removes this resource.
4. A site editor does something else, causing items tagged a to be cleared.

In this case we want to remove http://www.example.com/foo from the cache in step three. If we're not able to remove this item from the tag store, however, it will be included in the list of items we're getting in step 4. One might think that it's not the end of the world if we're clearing a superfluous, non-existent resource, but in some cache implementations, you want to avoid that.

As for `untagItemId` and `getTagsForItemIds`, they're mostly there for completeness, but could potentially be useful for debugging and introspection.